documentation/asciidoc/computers/raspberry-pi/bcm2711-bootloader.adoc

```diff
@@ -280,6 +284,69 @@ Previously this property was only checked by the `rpi-eeprom-update` script. How
 
 Default: `0`
 
+[[HTTP_HOST]]
+==== HTTP_HOST
+
+If an HTTP download is initiated, `boot.img` and `boot.sig` are downloaded from this server.
+
+Invalid host names will be ignored. They should only contain lower case alphanumeric characters and `-` or `.`.
+If `HTTP_HOST` is set then HTTPS is disabled and plain HTTP used instead.
+You can specify an IP address to avoid the need for a DNS lookup.
+Don`t include the HTTP scheme or any forward slashes in the hostname.
+
+Default: `fw-download-alias1.raspberrypi.com`
+
+[[HTTP_PORT]]
+==== HTTP_PORT
+
+You can change the port used for HTTP downloads. This is ignored if `HTTP_HOST` is not set.
+
+By default network install and HTTP boot mode use port `443` because HTTPS is enabled when using the default host `fw-download-alias1.raspberrypi.com`. If `HTTP_HOST` is changed then HTTPS is not supported and HTTP will be used instead, so port `80` becomes the default.
+
+[[HTTP_PATH]]
+==== HTTP_PATH
+
+The path used for HTTP downloads.
+
+The case of the path *is* significant.
+Use forward (Linux) slashes for the path separator.
+Leading and trailing forward slashes are not required.
+
+If `HTTP_HOST` is not set, `HTTP_PATH` is ignored and the URL will be `\https://fw-download-alias1.raspberrypi.com:443/net_install/boot.img`. If `HTTP_HOST` is set the URL will be `\http://<HTTP_HOST>:<HTTP_PORT>/<HTTP_PATH>/boot.img`
+
+Default: `net_install`
+
+[[IMAGER_REPO_URL]]
+==== IMAGER_REPO_URL
+
+The imager application is configured with a json file downloaded on start up.
+
+You can change the URL of the json file used by the imager application to get it to offer your own images.
+You can test this with the Windows or Linux Imager applications by passing the URL via the `--repo` argument.
+
+Default: http://downloads.raspberrypi.org/os_list_imagingutility_v3.json
+
+[[NET_INSTALL_ENABLED]]
+==== NET_INSTALL_ENABLED
+
+By default network install is enabled on Raspberry Pi 4 and Raspberry Pi 400. On Compute Module 4 and other boards it is disabled by default.
+When network install is enabled it detects a keyboard and displays the network install screen on boot.
+
+To enable network install add, `NET_INSTALL_ENABLED=1`, or to disable network install add `NET_INSTALL_ENABLED=0`.
+
+This setting is ignored and network install / keyboard detection on boot is disabled if `DISABLE_HDMI=1` is set.
+
+[[NET_INSTALL_KEYBOARD_WAIT]]
+==== NET_INSTALL_KEYBOARD_WAIT
+
+If network install is enabled, the bootloader attempts to detect a keyboard and the `SHIFT` key to initiate network install. You can change the length of this wait in milliseconds with this property.
+
+Setting this to `0` disables the keyboard wait, although network install can still be initiated if no boot files are found and USB boot `4` is in `BOOT_ORDER`.
+
+NOTE: Testing suggests Keyboard / shift detection takes at least 750ms.
+
+Default: `900`
```

Comment:
Does this have a maximum value? Or could the user specify e.g. 10000? And if they did that, would it always wait 10 seconds for a SHIFT key to be pressed on a USB keyboard, even if there's a bootable OS on the SD card?

Reply:
No max value - maybe that's wrong?
It's a hard wait so it slows down boot and setting it to 10s would wait even when there's a valid bootable OS available

Reply:
That's kinda what I was expecting - thanks for the confirmation :+1: